### PR TITLE
fix: [PL-29635]: secret manager search

### PR DESCRIPTION
**Pull request by swarajcheguri-harness**
### Summary
Searching for the secret manager if not present while creating a new secret

#### Screenshots
[screen-capture (37).webm](https://user-images.githubusercontent.com/73115842/216015169-0b1a2dee-cac4-40d1-ba08-baf295dbd87b.webm)

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by swarajcheguri-harness** [comment]
retrigger build

File: src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx [hunk 1]
```diff
@@ -137,20 +147,58 @@ const CreateUpdateSecret: React.FC<CreateUpdateSecretProps> = props => {
...
-    queryParams: {
-      accountIdentifier,
-      orgIdentifier,
-      projectIdentifier,
-      category: 'SECRET_MANAGER',
-      source_category: sourceCategory
-    },
-    lazy: true
+    lazy: true,
+    debounce: 1000
```
Comment (avinashmadhwani02) [inline src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx:155]: Do we need 1 sec of debounce here? Can you check what are we using at other places? Let's be consistent

File: src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx [hunk 2]
```diff
@@ -137,20 +147,58 @@ const CreateUpdateSecret: React.FC<CreateUpdateSecretProps> = props => {
...
   })
+  useEffect(() => {
+    try {
+      getConnectorListPromise({
```
Comment (avinashmadhwani02) [inline src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx:159]: Can't we call this API as a hook? we will not have to save the data in state explicitly.

File: src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx [hunk 3]
```diff
@@ -137,20 +147,58 @@ const CreateUpdateSecret: React.FC<CreateUpdateSecretProps> = props => {
...
+        queryParams: {
+          accountIdentifier,
+          orgIdentifier,
+          projectIdentifier,
+          category: 'SECRET_MANAGER',
+          source_category: sourceCategory
+        }
+      }).then(data => {
+        setSecretManagersApiResponse(data)
+        const defaultSecretManagerIdLocal = data?.data?.content?.filter(item => item.connector?.spec?.default)?.[0]
```
Comment (avinashmadhwani02) [inline src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx:169]: Can we split this into multiple lines? 
difficult to read

File: src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx [hunk 4]
```diff
@@ -137,20 +147,58 @@ const CreateUpdateSecret: React.FC<CreateUpdateSecretProps> = props => {
...
+        setInitialSecretManagerAPICallInProgress(false)
+      })
+    } catch (error) {
+      modalErrorHandler?.showDanger(getErrorInfoFromErrorObject(error))
+    }
+  }, [])
+
+  useEffect(() => {
+    if (lazySecretManagersApiResponse) {
+      setSecretManagersApiResponse(lazySecretManagersApiResponse)
```
Comment (avinashmadhwani02) [inline src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx:184]: Why are we setting this in the state?

File: src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx (hunk 3 above)
Comment (mayankVermaHarness) [reply src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx:169]: If using 0th item from the result of `Array.filter`, consider using `Array.find` instead

File: src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx (hunk 1 above)
Comment (swarajcheguri-harness) [reply src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx:155]: added 500 ms like in many places in NGUI

**Review by mayankVermaHarness — DISMISSED** [review]

File: src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx [hunk 5]
```diff
@@ -148,9 +156,37 @@ const CreateUpdateSecret: React.FC<CreateUpdateSecretProps> = props => {
...
+          item => item.connector?.spec?.default
+        )?.connector?.identifier
+        if (defaultSecretManagerIdLocal) {
+          setDefaultSecretManagerId(defaultSecretManagerIdLocal)
+        }
+      }
+    }
+  }, [secretManagersApiResponse])
+
+  useEffect(() => {
```
Comment (avinashmadhwani02) [inline src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx:176]: This useEffect is not required, you can directly pass search_term in hook query params

File: src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx [hunk 6]
```diff
@@ -148,9 +156,37 @@ const CreateUpdateSecret: React.FC<CreateUpdateSecretProps> = props => {
       category: 'SECRET_MANAGER',
       source_category: sourceCategory
     },
-    lazy: true
+    debounce: 500
   })
 
+  useEffect(() => {
+    if (secretManagersApiResponse) {
```
Comment (avinashmadhwani02) [inline src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx:163]: Please add error handling

**Review by avinashmadhwani02 — APPROVED** [review]

**Review by mayankVermaHarness — APPROVED** [review]

**Pull request merged**